mechanics/RELEASE-VERSIONING-PRINCIPLES.md

```diff
@@ -0,0 +1,727 @@
+qualified. We expect this to work given Kubernetes backwards compatibility
+support. We intend for these upgrades to be a non-issue. If particular versions
+are found to be incompatible and the Knative version is still community
+supported we will either: (1) work with the Kubernetes community on a fix or (2)
+develop a patch release to workaround the issue. We will publish guidance on
+potential incompatibility on knative.dev
+
+### Which API endpoints should I enable when I vend/host Knative Serving?
+
+Vendors may make their decisions about supporting API versions. They may decide
+to support optional API versions by default or may support API versions that are
+EOL in the open source community. The recommendation of the community is to
+offer all APIs versions that the Open Source project makes available by default
+for a given release version. This offers clients the best interoperability and
+users the best portability.
+
+### What about Serving v1alpha1 API support?
+
+Due to the length of time we have supported v1alpha1, the ubiquity of
+availability across released Knative Serving versions, and the presence of
+clients statically developed against this version we will treat the current set
+of v1alpha1 Serving APIs as 'Beta' APIs. This means that we will keep v1alpha1
+available in releases for at least 9 months after deprecated is announced.
+During that period of time of availability, we will evaluate whether the API is
+available through the default Open Source installation by following the _Default
+and optional API versions principle._
+
+### What happens when the minimum Kubernetes version for a supported Knative release is no longer supported by Kubernetes?
+
+We will explicitly avoid introducing changes into a Knative Serving release
+branch that would raise the minimum version. We will not keep around test
+clusters for unsupported Kubernetes versions, but we will attempt to solve user
+reported bugs to the best of our ability.
+
+### Can we have the table below on Knative.dev?
+
+Yes. We will publish a table similar to below once we have agreement on the
+principles so that users can easily determine versions and endpoints available
+for a given release.
+
+## Knative Serving Version Table
+
+Storage Type = The type that is stored in etcd by the Kubernetes API Server
+
+API Endpoints = The rest endpoints that are made available to clients
+
+API Types = The API types that the endpoint understands
+
+Lemonade = v1alpha1 + v1beta1 super-set type
+
+Force Upgrade = Automatically convert v1alpha1 to v1beta1 syntax via Lemonade
+
+Availability =
+
+- Default means installed through typical open source release.
+- Optional means available in the open source, but will not be installed by
+  default.
+- Removed means support no longer exists within the open source release
+
+**Note**: subject to change based on verification of (knative, kube) version
+tuple
+
+<table>
```

Comment:
This table it pretty serving specific, which I could see causing confusion out of that context.

Reply:
Does line 174 help with that? Should I repeat it again here?

Reply:
yep, I missed that.  thanks.